### DISCUSSION TITLE
Mysql query issue, backslashes escaping Json Collection/Objects

### DISCUSSION THREAD

**Discussion by teksrq** [livewire/livewire#4887] [1. Help]
Livewire is performing awesome, my issue is an Eloquent query in a Livewire component. I have two similar queries, both returning a Collection of results with a column containing a JSON collection of JSON objects. The method that came with the Livewire project component doesn't add any escapes/backslashes, but the query I'm attempting in the same component returns a similar column where all quotes escaped. I've spent a few days trying to figure out how to remove/prevent the backslashes in the results prior to receiving the Collection, nothing I've tried works. All I can do is extract the escaped JSON string from each individual array in the Collection, process out the backslashes, and then convert each string back to an associative array. Seems crazy, but that's all I've had success with...

I've spent some time trying to figure out how the Livewire method(s) accomplish this, but I can't find the magic. The component method throws the query string to DataTables for further processing, from there I can't find the endpoint (meaning I can't find the final **get()** trigger). 

Here's the method from the component:
```
public function getRowsQueryProperty()
    {
        $query = Project::query()
            ->select('projects.*')
            ->when($this->filters['status'], fn($query, $status) => $query->where('status', $status))
            ->when($this->filters['date-min'], fn($query, $date) => $query->where('start_date', '>=', Carbon::parse($date)))
            ->when($this->filters['date-max'], fn($query, $date) => $query->where('start_date', '<=', Carbon::parse($date)))
            ->when($this->filters['search'], fn($query, $search) => $query->where('name', 'like', '%'.$search.'%'))
            ->addSelect(['teams' => ProjectTeam::selectRaw('
                JSON_ARRAYAGG(
                    JSON_OBJECT(
                        "id", `project_team`.`id`,
                        "name", `teams`.`name`
                    )
                ) AS teams
            ')
                ->join('teams', 'teams.id', '=', 'project_team.team_id', 'inner')
                ->whereColumn('project_id', 'projects.id')
                ->groupBy('project_id')
                ->orderBy('teams.name')
        ]);

        return $this->applySorting($query);
    }
```
and the Json results are easy to parse:
```
"teams" => "[{"id": 98, "name": "Jane and Charlie"},{"id": 99, "name": "Mary and Jane"}]"
```
Here's my query in the same component that is adding backslashes to the results:
```
$query = ProjectTeam::query()
            ->select('project_team.project_id')
            ->selectRaw('
                JSON_ARRAYAGG(
                    JSON_OBJECT(
                        "team_id",
                        `teams`.`id`,
                        "team",
                        teams.name,
                        "id",
                        `users`.`id`,
                        "name",
                        `users`.`name`
                    )
                ) AS teams_users
            ')
            ->join('teams', 'teams.id', '=', 'project_team.team_id', 'inner')
            ->join('team_user', 'project_team.team_id', '=', 'team_user.team_id')
            ->join('users', 'team_user.user_id', '=', 'users.id')
            ->whereIn('project_team.project_id', $ids)
            ->groupBy('project_team.project_id')
            ->get()
            ->keyBy('project_id');
```
And the escaped results I can't easily work with:
```
"teams_users" => "["{\"team_id\": 8, \"team\": \"Dad\", \"id\": 1, \"name\": \"Dad\"}"]"
```
I feel like it has to be something simple, but I can't find out why it's adding slashes, been tweaking at it for a few days.
Does anyone know how I can avoid them in the first place? Query methods such as **toJson()** have no effect.
(it likely goes without saying that I'm only two months into my Laravel/Livewire journey, I'm very new to the framework.)

**Comment by teksrq** [comment]
Livewire (or something in the base project) is definitely (correctly) filtering the backslashes out somewhere. I found a temporary work around by modifying the query so mysql doesn't escape the quotes, using CONCATs:
```
->selectRaw('
                CONCAT(
                  "[",
                  GROUP_CONCAT(
                    JSON_OBJECT(
                        "team_id",
                        `teams`.`id`,
                        "team",
                        teams.name,
                        "id",
                        `users`.`id`,
                        "name",
                        `users`.`name`
                        )
                    ),
                    "]"
                ) AS teams_users
            ')
```
This spits out usable variables, unless there's a quotation mark anywhere in the names, mysql still escapes those. Then I'm back in a similar situation, the name is truncated at the first quotation mark. I really need to figure out how it's handling quotes correctly...

**Comment by DavisOsiemo** [comment]
Hi there, did you find a fix for this?

**Reply by teksrq** [reply]
Going back and looking at the old code (last edit was 04/22), I did not. It was a learning project for me, so I stopped working on it once I got busy.
